music_assistant/providers/musiccast/__init__.py

```diff
@@ -0,0 +1,840 @@
+                player.state = PlayerState.IDLE
+
+        # SOURCES
+        player.source_list = UniqueList([])
+        for source_id, source_name in device.source_mapping.items():
+            control = source_id in MC_CONTROL_SOURCE_IDS
+            passive = source_id in MC_PASSIVE_SOURCE_IDS
+            player.source_list.append(
+                PlayerSource(
+                    id=source_id,
+                    name=source_name,
+                    passive=passive,
+                    can_play_pause=control,
+                    can_seek=False,
+                    can_next_previous=control,
+                )
+            )
+
+        # UPDATE UPNP HELPER
+        update_helper = self.upnp_update_helper.get(player.player_id)
+        now = time.time()
+        queue = self.mass.player_queues.get_active_queue(player.player_id)
+        if update_helper is None or now - update_helper.last_poll > 5:
+            # Let's not do this too often
+            # verify, that we are playing
+            try:
+                _xml_media_info = await avt_get_media_info(
+                    self.mass.http_session, device.physical_device
+                )
+            except ServerDisconnectedError:
+                return
+            _player_current_url = search_xml(_xml_media_info, "CurrentURI")
+            _queue_item_id = search_didl_queueitemid(_xml_media_info)
+
+            # controlled by mass is only True, if we are directly controlled
+            # i.e. we are not a group member.
+            # the device's source id is server, if controlled by upnp, but also, if the internal
+            # dlna function of the device are used. As a fallback, we then
+            # use the item's title. This can only fail, if our current and next item
+            # has the same name as the external.
+            controlled_by_mass = False
+            if (
+                queue is not None
+                and queue.current_item is not None
+                and queue.current_item.media_item is not None
+                and _player_current_url is not None
+            ):
+                _item_ids = [queue.current_item.queue_item_id]
+                _names = [queue.current_item.media_item.name]
+                if queue.next_item is not None and queue.next_item.media_item is not None:
+                    _item_ids.append(queue.next_item.queue_item_id)
+                    _names.append(queue.next_item.media_item.name)
+                controlled_by_mass = (
+                    player.player_id in _player_current_url
+                    and _queue_item_id in _item_ids
+                    and device.source_id == "server"
+                    # this last one might fail in the rare case, that the external controller
+                    # is using the same title and its part of current/ next of our queue
+                    and device.media_title in _names
+                )
```

Comment:
```suggestion
           controlled_by_mass = player.player_id in _player_current_url and self.mass.streams.base_url in _player_current_url
```

all this needs to be removed, you need to tell the queue controller what is playing - not the other way around!